src/__tests__/resources/timeSeriesList.unit.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { TimeSeriesList } from '../../resources/classes/timeSeriesList';
+import { randomInt, setupLoggedInClient } from '../testUtils';
+
+describe('TimeSeriesList class unit test', async () => {
+  let axiosMock: MockAdapter;
+  let client: CogniteClient;
+  let newAsset: any;
+  let createdTimeSeries: TimeSeriesList;
+  let timeSeriesWithAssetId: any;
+  let createdAssets: AssetList;
+  let datapointArray: any[];
+  let assetsArray: any[] = [];
+  let timeseriesArray: any[] = [];
+  beforeAll(() => {
+    client = setupLoggedInClient();
+    axiosMock = new MockAdapter(client.instance);
+  });
+  beforeEach(async () => {
+    axiosMock.reset();
+
+    // Create assets
+    assetsArray = [];
+    for (let index = 0; index < 5; index++) {
+      newAsset = {
+        name: 'test-asset' + randomInt(),
+        externalId: 'asset' + randomInt(),
+        id: randomInt(),
+      };
+      assetsArray.push(newAsset);
+    }
+    axiosMock
+      .onPost(new RegExp('/assets$'), { items: assetsArray })
+      .replyOnce(200, { items: assetsArray });
+    createdAssets = await client.assets.create(assetsArray);
+
+    // Create timeseries
+    timeseriesArray = [
+      {
+        name: 'test-timeseries',
+        externalId: 'timeseries' + randomInt(),
+        id: randomInt(),
+      },
+    ];
+    for (let index = 0; index < 5; index++) {
+      timeSeriesWithAssetId = {
+        name: 'test-timeseries' + randomInt(),
+        assetId: createdAssets[index].id,
+        id: randomInt(),
+      };
+      timeseriesArray.push(timeSeriesWithAssetId);
+    }
+    axiosMock
+      .onPost(new RegExp('/timeseries$'), {
+        items: timeseriesArray,
+      })
+      .replyOnce(200, { items: timeseriesArray });
+    createdTimeSeries = await client.timeseries.create(timeseriesArray);
+
+    // Create datapoints
+    datapointArray = [];
+    const time = randomInt();
+    for (let index = 0; index < 3; index++) {
+      const datapoint = {
+        id: createdTimeSeries[index].id,
+        datapoints: [{ timestamp: time + index, value: 10 + index }],
+      };
+      datapointArray.push(datapoint);
+    }
+    axiosMock
+      .onPost(new RegExp('/timeseries/data$'), {
+        items: datapointArray,
+      })
+      .replyOnce(200);
+    await client.datapoints.insert(datapointArray);
+  });
+
+  test('delete', async () => {
+    const timeSeriesIds = createdTimeSeries.map(timeseries => ({
+      id: timeseries.id,
+    }));
+    axiosMock
+      .onPost(new RegExp('/timeseries/delete$'), {
+        items: timeSeriesIds,
+      })
+      .replyOnce(200, {});
+    await createdTimeSeries.delete();
+  });
+
+  test('get all assets', async () => {
+    const assetIds = createdTimeSeries
+      .map(timeseries => ({
+        id: timeseries.assetId,
+      }))
+      .filter(assetId => assetId.id !== undefined);
+    axiosMock
+      .onPost(new RegExp('/assets/byids$'), { items: uniqBy(assetIds, 'id') })
```

Comment:
I thought the second argument to `uniqBy` had to be a function 🤔

Reply:
![image](https://user-images.githubusercontent.com/37441744/62679516-feb46d80-b9b4-11e9-9b19-af4d70ba6870.png)

Could be like this as well